luarules/gadgets/unit_area_timed_damage.lua

```diff
@@ -12,484 +12,229 @@ function gadget:GetInfo()
 end
 
 if not gadgetHandler:IsSyncedCode() then
-    return
+	return
 end
 
------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+---------------------------------------------------------------------------------------------------------------
+--
+-- Unit setup -- ! todo: Do this for all the units with area timed damage
+--
+-- (1) CustomParams. Add these properties to the unit (for ondeath) or weapon (for onhit).
+--
+-- area_timed_damage  -  true            -  Required.
+--    atd_duration    -  <number>        -  Required. The duration of the effect.
+--    atd_stable_ceg  -  <string> | nil  -  Name of the CEG that appears continuously for the duration.
+--    atd_damage_ceg  -  <string> | nil  -  Name of the CEG that appears on anything damaged by the effect.
+--    atd_resistance  -  <string> | nil  -  The 'type' of the damage, which can be resisted. Be uncreative.
```

Comment:
Please make code as straightforward as possible, do not obfuscate using acronyms, short forms, etc. Anyone reading the code needs to parse what `atd` stands for.